pkg/credentialprovider/provider.go

```diff
@@ -83,7 +83,7 @@ func (d *defaultDockerConfigProvider) Enabled() bool {
 // Provide implements dockerConfigProvider
 func (d *defaultDockerConfigProvider) Provide() DockerConfig {
 	// Read the standard Docker credentials from .dockercfg
-	if cfg, err := ReadDockerConfigFile(); err == nil {
+	if cfg, err := ReadDockerConfigFile(nil); err == nil {
```

Comment:
so to insert your own search dir, while keeping the fallback dirs, you would do this?

```
paths := []string{mypath}
paths = append(paths, DefaultDockerConfigJSONPaths()...)
paths = append(paths, DefaultDockercfgPaths()...)
cfg, err := ReadDockerConfigFile(paths)
```

it means you could load a config.json from one of the default dockercfg paths, or vice-versa... that seems off to me, but maybe it's not a big deal